qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -638,16 +638,15 @@ def _run_analysis(
         # to generate the figure
         plot_bool = plot == "always" or (plot == "selective" and quality == "bad")
 
-        if self.options.return_fit_parameters:
-            # Store fit status overview entry regardless of success.
-            # This is sometime useful when debugging the fitting code.
-            overview = AnalysisResultData(
-                name=PARAMS_ENTRY_PREFIX + self.name,
-                value=fit_data,
-                quality=quality,
-                extra=self.options.extra,
-            )
-            result_data.append(overview)
+        # Store fit status overview entry regardless of success.
+        # This is sometime useful when debugging the fitting code.
+        overview = AnalysisResultData(
```

Comment:
Why do we keep this object? Now it's stored in the artifact right? Is this due to the index change of the analysis results? I would turn default option value into `None` and issue the deprecation warning unless user doesn't explicitly set this to `False` (maybe too annoying though). With this change we'll lose the knob to trigger deprecation warning in future when we remove this object, i.e. we cannot raise the warning when Python list `[0]` is called for the analysis results.

Reply:
We had some discussions last release about how to deprecate/remove this entry (https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1342#discussion_r1481833109). I think since we've had one release cycle with artifacts it's ok to remove this directly. What do you think @wshanks?

Reply:
The reason we kept this before was because many tests used integer arguments to `analysis_results()` and I worried that that likely reflected a common pattern in user code as well. So removing the first result would change the integer indices of all other results.

I don't think it is that big of a burden to leave the overview code in for now. I think I would leave it in along with `return_fit_parameters` until either passing an integer or slice or passing `dataframe=False` to `analysis_results()` is removed. What is the timeline for those removals?

Reply:
Maybe deprecation for 
- When search index is int type or None, and
- When the result includes overview entry

it the reasonable approach?

Reply:
@nkanazawa1989 we are already issuing deprecating warnings for int/slice search index and retrieving the overview entry: https://github.com/Qiskit-Extensions/qiskit-experiments/blob/fe4ccff973dae85fdff3416b253880051b71e6c3/qiskit_experiments/framework/experiment_data.py#L1569-L1582
Do you want to implement this differently?

Reply:
Sorry I missed it. Since this deprecation is in 0.6 in principle we can remove the entry in this release?

Reply:
We can remove it now based on the deprecation policy, but what about my comment above?

> I don't think it is that big of a burden to leave the overview code in for now. I think I would leave it in along with return_fit_parameters until either passing an integer or slice or passing dataframe=False to analysis_results() is removed. What is the timeline for those removals?

Reply:
Yeah, I'm fine with leaving it as is until we're ready to make the full change since artifacts (and figures) are also currently accessible via the integer key. I think it's awkward for the behavior to be different across these interfaces. What do you think @nkanazawa1989? Maybe we just deprecate retrieving artifacts by index in this release?

Reply:
Sounds like a plan. Let's deprecate it as well in this release and remove all of them in 0.8.

Reply:
@coruscating I see you added the deprecation. Can we also keep put back the `return_fit_parameters` option until we remove returning the overview completely?

Reply:
Done, this is ready for another look.

Reply:
The part touched on here looks good now. The rest of the PR looks good as well but I didn't review very closely and will leave it to @nkanazawa1989.